test/add-member.test.js

```diff
@@ -272,3 +273,177 @@ test('addMembers too many members', async (t) => {
 
   await p(alice.close)(true)
 })
+
+test('addMembers adds to all the tip epochs and gives keys to all the old epochs as well', async (t) => {
+  // alice adds bob and carol
+  // alice and bob remove carol at the same time, creating forked epochs
+  // everyone still replicates and sees the fork
+  // alice adds david to the group, and he should see both forks and the original epoch
+  const alice = Testbot({
+    keys: ssbKeys.generate(null, 'alice'),
+    mfSeed: Buffer.from(
+      '000000000000000000000000000000000000000000000000000000000000a1ce',
+      'hex'
+    ),
+  })
+  const bob = Testbot({
+    keys: ssbKeys.generate(null, 'bob'),
+    mfSeed: Buffer.from(
+      '0000000000000000000000000000000000000000000000000000000000000b0b',
+      'hex'
+    ),
+  })
+  const carol = Testbot({
+    keys: ssbKeys.generate(null, 'carol'),
+    mfSeed: Buffer.from(
+      '00000000000000000000000000000000000000000000000000000000000ca201',
+      'hex'
+    ),
+  })
+  const david = Testbot({
+    keys: ssbKeys.generate(null, 'david'),
+    mfSeed: Buffer.from(
+      '00000000000000000000000000000000000000000000000000000000000da71d',
+      'hex'
+    ),
+  })
+
+  await Promise.all([
+    alice.tribes2.start(),
+    bob.tribes2.start(),
+    carol.tribes2.start(),
+    david.tribes2.start(),
+  ])
+
+  const [, bobRootId, carolRootId, davidRootId] = (
+    await Promise.all(
+      [alice, bob, carol, david].map((peer) => p(peer.metafeeds.findOrCreate)())
+    )
+  ).map((root) => root.id)
```

Comment:
shit, I though you had to do `.then(roots => roots.map(r => r.id))`

is it the magic of `(await Promise)` which lets you just slap a map on the end there?

Reply:
yeah the await just turns it into a normal array